apps/explorer/src/utils/getAmount.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+    getPaySuiTransaction,
+    getPayTransaction,
+    getTransferSuiAmount,
+} from '@mysten/sui.js';
+
+import type { SuiTransactionKind } from '@mysten/sui.js';
+
+// TODO: Move this to sui.js
+export function getAmount(txnData: SuiTransactionKind): number | bigint | null {
+    const paySuiData =
+        getPaySuiTransaction(txnData) ?? getPayTransaction(txnData);
+    // Sum Sui Pay Array Amounts
+    const paySuiAmount = paySuiData?.amounts.reduce(
+        (acc, value) => value + acc,
+        0
+    );
```

Comment:
I think you should aggregate the amounts by recipients here. `PaySui`, `Pay`, and `PayAllSui` can all have multiple recipients, and we probably want to breakdown the amounts by recipient instead of showing the total for all recipients. cc @Jordan-Mysten @mystie711 @bwann52

Reply:
I see, I'll update it.

Reply:
Spoke with Godwin and realized i misunderstood this specific response. We all good.